services/graph/pkg/service/v0/educationuser.go

```diff
@@ -0,0 +1,481 @@
+			render.Status(r, http.StatusInternalServerError)
+			render.JSON(w, r, user)
+			return
+		}
+		if lspr.GetStatus().GetCode() != cs3rpc.Code_CODE_OK {
+			logger.Debug().Str("grpc", lspr.GetStatus().GetMessage()).Msg("could not get drive for education user")
+			// in case of NOT_OK, we can just return the user object with empty drives
+			render.Status(r, status.HTTPStatusFromCode(http.StatusOK))
+			render.JSON(w, r, user)
+			return
+		}
+		drives := []libregraph.Drive{}
+		for _, sp := range lspr.GetStorageSpaces() {
+			d, err := g.cs3StorageSpaceToDrive(r.Context(), wdu, sp)
+			if err != nil {
+				logger.Debug().Err(err).Interface("id", sp.Id).Msg("error converting space to drive")
+				continue
+			}
+			quota, err := g.getDriveQuota(r.Context(), sp)
+			if err != nil {
+				logger.Debug().Err(err).Interface("id", sp.Id).Msg("error calling get quota on drive")
+			}
+			d.Quota = quota
+			if slices.Contains(sel, "drive") || slices.Contains(exp, "drive") {
+				if *d.DriveType == "personal" {
+					user.Drive = d
+				}
+			} else {
+				drives = append(drives, *d)
+				user.Drives = drives
+			}
+		}
+	}
+
+	render.Status(r, http.StatusOK)
+	render.JSON(w, r, user)
+}
+
+// DeleteEducationUser implements the Service interface.
+func (g Graph) DeleteEducationUser(w http.ResponseWriter, r *http.Request) {
+	logger := g.logger.SubloggerWithRequestID(r.Context())
+	logger.Info().Msg("calling delete education user")
+	userID := chi.URLParam(r, "userID")
+	userID, err := url.PathUnescape(userID)
+	if err != nil {
+		logger.Debug().Err(err).Str("id", userID).Msg("could not delete education user: unescaping education user id failed")
+		errorcode.InvalidRequest.Render(w, r, http.StatusBadRequest, "unescaping education user id failed")
+		return
+	}
+
+	if userID == "" {
+		logger.Debug().Msg("could not delete education user: missing education user id")
+		errorcode.InvalidRequest.Render(w, r, http.StatusBadRequest, "missing education user id")
+		return
+	}
+	logger.Debug().Str("id", userID).Msg("calling get education user on user backend")
+	user, err := g.identityEducationBackend.GetEducationUser(r.Context(), userID, r.URL.Query())
+	if err != nil {
+		logger.Debug().Err(err).Str("userID", userID).Msg("failed to get education user from backend")
+		var errcode errorcode.Error
+		if errors.As(err, &errcode) {
+			errcode.Render(w, r)
+		} else {
+			errorcode.GeneralException.Render(w, r, http.StatusInternalServerError, err.Error())
+		}
+		return
+	}
+
+	e := events.UserDeleted{UserID: user.GetId()}
+	if currentUser, ok := revactx.ContextGetUser(r.Context()); ok {
+		if currentUser.GetId().GetOpaqueId() == user.GetId() {
+			logger.Debug().Msg("could not delete education user: self deletion forbidden")
+			errorcode.NotAllowed.Render(w, r, http.StatusForbidden, "self deletion forbidden")
+			return
+		}
+		e.Executant = currentUser.GetId()
+	}
+
+	if g.gatewayClient != nil {
```

Comment:
Is it a valid option to run graph service without gateway client? Imho it is a must have so this check is not needed

Reply:
Yes. It's possible to run it standalone. That's actually how we want to run it for the education provisioning service. https://github.com/owncloud-cst/koko-deployment/pull/83#issue-1496784865 has some hints on how to do that.